crates/turborepo-lib/src/run/summary/mod.rs

```diff
@@ -122,50 +198,174 @@ impl<'a> RunSummary<'a> {
             RunType::Real
         };
 
-        let execution_summary = ExecutionSummary::new(
-            synthesized_command.clone(),
-            package_inference_root,
-            start_at,
-        );
-
-        RunSummary {
-            inner: RunSummaryInner {
-                id: Ksuid::new(None, None),
-                version: RUN_SUMMARY_SCHEMA_VERSION.to_string(),
-                execution_summary: Some(execution_summary),
-                turbo_version: turbo_version.to_string(),
-                packages,
-                env_mode: run_opts.env_mode,
-                framework_inference: run_opts.framework_inference,
-                tasks: vec![],
-                global_hash_summary,
-                scm: SCMState::get(&env_at_execution_start, repo_root),
-                user: get_user(&env_at_execution_start),
-                monorepo: !single_package,
-            },
-            run_type,
+        let execution_summary = self
+            .execution_tracker
+            .finish(package_inference_root, exit_code, start_time, end_time)
+            .await?;
+
+        Ok(RunSummary {
+            id: Ksuid::new(None, None),
+            version: RUN_SUMMARY_SCHEMA_VERSION.to_string(),
+            turbo_version: turbo_version.to_string(),
+            packages,
+            execution: Some(execution_summary),
+            env_mode: run_opts.env_mode.into(),
+            framework_inference: run_opts.framework_inference,
+            tasks: vec![],
+            global_hash_summary,
+            scm: self.scm,
+            user: self.user,
+            monorepo: !single_package,
             repo_root,
-            single_package,
             should_save,
+            run_type,
+            spaces_client_handle: self.spaces_client_handle,
+        })
+    }
+
+    #[tracing::instrument(skip(pkg_dep_graph, ui))]
+    pub async fn finish<'a>(
+        self,
+        start_time: DateTime<Local>,
+        exit_code: u32,
```

Comment:
Does the Spaces API require an unsigned int? [Rust](https://doc.rust-lang.org/std/process/struct.ExitStatus.html#method.code) and [Go](https://pkg.go.dev/os#ProcessState.ExitCode) produce signed ints if so we should probably port whatever conversion logic we had in Go

```suggestion
        exit_code: i32,
```

Reply:
Just an assumption on my part. Changed